crates/derive/src/online/alloy_providers.rs

```diff
@@ -60,6 +60,17 @@ impl AlloyChainProvider {
         let inner = ReqwestProvider::new_http(url);
         Self::new(inner)
     }
+
+    /// Returns the chain ID.
+    pub async fn chain_id(&mut self) -> Result<u64> {
+        let chain_id: TransportResult<alloc::string::String> =
+            self.inner.raw_request("eth_chainId".into(), ()).await;
+        let chain_id = match chain_id {
```

Comment:
Not necessary, but you can use `alloy`'s `U64` type to decode hex strings rather than manually stripping the prefix

Reply:
Good suggestion - will put up a ticket for it :)